models/aeif_cond_alpha.cpp

```diff
@@ -280,6 +280,17 @@ nest::aeif_cond_alpha::Parameters_::set( const DictionaryDatum& d )
 
   if ( Delta_T != 0. && V_peak_ <= V_th )
     throw BadProperty( "V_peak must be larger than threshold." );
+  else if ( Delta_T != 0. )
+  {
+    // check for possible numerical overflow with the exponential divergence at
+    // spike time
+    double max_exp_arg = std::log( std::numeric_limits< double >::max() );
+    if ( ( V_peak_ - V_th ) / Delta_T >= max_exp_arg )
+      throw BadProperty(
+        "The current combination of V_peak, V_th and Delta_T \
+will lead to numerical overflow at spike time; try for instance to increase \
+Delta_T or to reduce V_peak to avoid this problem." );
+  }
   else if ( Delta_T == 0. )
     updateValue< double >( d, names::V_peak, V_th ); // expected behaviour
```

Comment:
Yes, but the coding here is unfortunate. On lines 258-259, we extract `V_peak` and `V_th` using `updateValue()`. If we now need to set `V_peak` to `V_th`, that should be done by direct assignment, not by calling `updateValue()` again. And strictly speaking, we should not change `V_th` (since that is a user-defined parameter). Rather, we should introduce an additional variable in `V_` that represents the `V_th` to use, and that is set in `calibrate()`:
```C++
if ( P_.Delta_T > 0 )
{
  V_.V_th = P_.V_th;
}
else
{
  V_.V_th = P_.V_peak;
}
```
(I hope I got the logic the right way around?)

Then the user could even switch back and forth between Delta_T=0 and > 0 and all would work as expected.